tests/utils/node/nodes.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/config/toml"
+	"github.com/ChainSafe/gossamer/tests/utils/config"
+)
+
+// Nodes is a slice of nodes.
+type Nodes []Node
+
+// MakeNodes creates `num` nodes using the `baseNode`
+// as a base for each node. It sets the following fields:
+// - the first node is always the BABE lead
+// - the index of each node is incremented per node
+// - the base path is set to a test temporary directory
+// - remaining unset fields are set to their default.
+func MakeNodes(t *testing.T, num int, tomlConfig toml.Config,
+	options ...Option) (nodes Nodes) {
+	nodes = make(Nodes, num)
+	for i := range nodes {
+		nodes[i].tomlConfig = tomlConfig
+		// Set fields using options given
+		for _, option := range options {
+			option(&nodes[i])
+		}
+
+		// Set defaults using index `i`
+		nodes[i].tomlConfig.Core.BABELead = i == 0
+
+		if nodes[i].index == nil {
+			nodes[i].index = intPtr(i)
+		}
+
+		// Set node defaults on the remaining unset fields
+		nodes[i].setDefaults(t)
```

Comment:
should this use `node.NewNode`

Reply:
Done 👍 
Now it always sets each node to have their 'index' as the index of the nodes slice, so the `SetIndex` option is ignored. We don't use anywhere so all good as well.